lightning/src/util/persist.rs

```diff
@@ -249,10 +259,1010 @@ where
+			}
+		}
+	}
+
+	/// Convert this monitor update name to a &str
+	pub fn as_str(&self) -> &str {
+		&self.1
+	}
+
+	/// Left-pad the sequential update id for string representation
+	fn left_padded_string(value: u64) -> String {
+		format!("{:0>20}", value)
+	}
+}
+
+impl From<u64> for UpdateName {
+	fn from(value: u64) -> Self {
+		Self(value, UpdateName::left_padded_string(value))
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::chain::chainmonitor::Persist;
+	use crate::chain::ChannelMonitorUpdateStatus;
+	use crate::events::{ClosureReason, MessageSendEventsProvider};
+	use crate::ln::functional_test_utils::*;
+	use crate::util::test_utils::{self, TestLogger, TestStore};
+	use crate::{check_added_monitors, check_closed_broadcast};
+
+	const EXPECTED_UPDATES_PER_PAYMENT: u64 = 5;
+
+	#[test]
+	fn converts_u64_to_update_name() {
+		assert_eq!(UpdateName::from(0).as_str(), "00000000000000000000");
+		assert_eq!(UpdateName::from(21).as_str(), "00000000000000000021");
+		assert_eq!(UpdateName::from(u64::MAX).as_str(), "18446744073709551615");
+	}
+
+	#[test]
+	fn bad_update_name_fails() {
+		assert!(UpdateName::new("deadbeef".to_string()).is_err());
+		assert!(UpdateName::new("-0000000000000000001".to_string()).is_err());
+	}
+
+	#[test]
+	fn monitor_from_outpoint_works() {
+		let monitor_name1 = MonitorName::from(OutPoint {
+			txid: Txid::from_hex(
+				"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
+			)
+			.unwrap(),
+			index: 1,
+		});
+		assert_eq!(
+			monitor_name1.as_str(),
+			"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_1"
+		);
+
+		let monitor_name2 = MonitorName::from(OutPoint {
+			txid: Txid::from_hex(
+				"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef",
+			)
+			.unwrap(),
+			index: u16::MAX,
+		});
+		assert_eq!(
+			monitor_name2.as_str(),
+			"f33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeeff33dbeef_65535"
+		);
+	}
+
+	#[test]
+	fn bad_monitor_string_fails() {
+		assert!(MonitorName::new(
+			"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef".to_string()
+		)
+		.is_err());
+		assert!(MonitorName::new(
+			"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_65536".to_string()
+		)
+		.is_err());
+		assert!(MonitorName::new(
+			"deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef_21".to_string()
+		)
+		.is_err());
+	}
+
+	#[test]
+	fn persister_with_real_monitors() {
+		let test_max_pending_updates = 7;
+		let chanmon_cfgs = create_chanmon_cfgs(4);
+		let persister_0 = MonitorUpdatingPersister {
+			kv_store: &TestStore::new(false),
+			logger: &TestLogger::new(),
+			maximum_pending_updates: test_max_pending_updates,
+			entropy_source: &chanmon_cfgs[0].keys_manager,
+			signer_provider: &chanmon_cfgs[0].keys_manager,
+		};
+		let persister_1 = MonitorUpdatingPersister {
+			kv_store: &TestStore::new(false),
+			logger: &TestLogger::new(),
+			maximum_pending_updates: test_max_pending_updates,
```

Comment:
Good suggestion. It actually revealed to me a subtle misunderstanding in the final assertions (one of the nodes has an update because its counterparty FCs, which was hiding because we iterated a multiple of the pending update window). I think that's good, I can also throw in a final cleanup and it should exercise everything a good bit.

Reply:
Done